### DISCUSSION TITLE
Training run keeps failing at high batch numbers

### DISCUSSION THREAD

**Discussion by mhochst** [talmolab/sleap#1240] [Help!] [answered]
hi there, I keep receiving this in my terminal when I attempt to run training, the only settings I changed from default were validation factor to .2 and batch size to 28. I am unsure if there is anything I can do to increase the available memory? and I would appreciate any assistance on what specifically is the error I am receiving. Thank you.
Computer specs
Intel(R) Xeon(R) W-1370P @ 3.60GHz   3.60 GHz
64gb RAM
NVIDIA Quadro RTX 4000
![image](https://user-images.githubusercontent.com/122642039/227591667-801cb06c-1041-4e02-afc4-496593c511f3.png)

**Comment by roomrys** [comment, marked as answer]
Hi @mhochst,

A higher batch size requires more memory. The error you are receiving is an OOM (Out of Memory) issue and is a limitation of your hardware. As you have likely noticed, lowering the batch size should avoid this memory issue. Is there any particular reason you require a batch size of 28?

Another thing you can try which should help lower memory usage is switching from a single animal to a top-down model. The top down model is more memory efficient because it takes a centered crop of the animal before attempting to find all it's body parts. Thus, if your animal is relatively small to the entire size of the video, you should see improvements on memory usage. However, with a multi-animal model, you will need to do some post-estimation clean-up (see Tracking: Cull to Target Instance Count) to ensure only a single-animal is predicted in each frame.

Let us know if that helped answer your question!

Thanks,
Liezl

**Reply by mhochst** [reply]
thank you for the response! i was following the advice of a colleague that a higher batch size would be better for training the model, they are working with Deep lab cut however so im unsure how much that translates over.

so instead ofsingle model i switch to multi animal top down? is there a way to adjust the crop of the animal? our chambers have some obstruction due to the nature of the research so i am unsure if i can move the blue square to the actual animal
![image](https://user-images.githubusercontent.com/122642039/227596812-89607075-e0ae-4331-93f5-4ddd7b8a3ba7.png)

**Reply by mhochst** [reply]
i ran it again as the top down model and right at the end during inference i received this 
![image](https://user-images.githubusercontent.com/122642039/227602208-b2cd66d4-06d9-4733-be62-0dfa6bb99997.png)

**Reply by roomrys** [reply]
Hi @mhochst,

This is a [known error](https://github.com/talmolab/sleap/issues/759#issuecomment-1148033566) #759 that occurs when you have a mix of color and grayscale videos in your project. You can use the Toggle Grayscale button in the Video dialog to individually toggle each video OR if you are on v1.3.0, you can use the Toggle Grayscale button to toggle the grayscale of all videos at once #1215.

Thanks,
Liezl
